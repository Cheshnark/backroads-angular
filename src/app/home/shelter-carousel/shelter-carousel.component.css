.slider-wrap {
  width: 100%;
  max-width: 300px;
  margin: 0 auto;

  overflow-x: auto;
}

.slider-main {
  margin-bottom: 2.5rem;
}

@media (width >= 768px) {
  .slider-wrap {
    max-width: 50%;
  }

  .slider-main {
    position: relative;
  }

  /* ---------- Slider button ---------- */
  button {
    position: sticky;
    top: 40%;

    padding: 1rem;

    color: #f9f4f6;
    background-color: #585757;
  }

  .left {
    left: 0;
  }

  .right {
    right: 0;
  }

  /* ---------- Scroll bar ---------- */
  ::-webkit-scrollbar {
    height: 7px;
  }

  ::-webkit-scrollbar-thumb {
    background: #c9c7c7;
  }

  ::-webkit-scrollbar-thumb:hover {
    background: #585757;
  }
}

@media (width >= 1280px) {
  .slider-wrap {
    max-width: 80%;
  }
}
