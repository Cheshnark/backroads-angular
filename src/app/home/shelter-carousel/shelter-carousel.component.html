<div>
  <div class='slider-wrap'>
    <div class='slider-main flex items-center gap-8'>
      <button class='left' (click)="onClickLeft()"><app-chevron-left /></button>
      @for (shelter of shelters; track $index + shelter.title) {
        <app-shelter-cards [shelter]='shelter'/>
      }
      <button class='right' (click)="onClickRight()"><app-chevron-right /></button>
    </div>
  </div>
</div>
