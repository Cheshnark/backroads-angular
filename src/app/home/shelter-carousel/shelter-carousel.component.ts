import { Component } from '@angular/core';
import { ShelterCardsComponent } from './shelter-cards/shelter-cards.component';
import { ChevronLeftComponent } from '../../../assets/icons/chevron-left/chevron-left.component';
import { ChevronRightComponent } from '../../../assets/icons/chevron-right/chevron-right.component';

@Component({
  selector: 'app-shelter-carousel',
  standalone: true,
  imports: [ShelterCardsComponent, ChevronLeftComponent, ChevronRightComponent],
  templateUrl: './shelter-carousel.component.html',
  styleUrl: './shelter-carousel.component.css'
})
export class ShelterCarouselComponent {
  // shelter variable is temporary, this is going to come from the DB.
  shelters = [
    {
      title: 'Cangas de Onís',
      body: 'Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.',
      services: {
        water: true,
        electricity: false,
        pet: false,
        bathroom: true,
        shower: true,
        internet: false,
        washClothes: true,
        pool: false,
        bar: false
      }
    },
    {
      title: 'Cangas de Onís',
      body: 'Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.',
      services: {
        water: true,
        electricity: false,
        pet: false,
        bathroom: true,
        shower: true,
        internet: false,
        washClothes: true,
        pool: false,
        bar: false
      }
    },
    {
      title: 'Cangas de Onís',
      body: 'Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.',
      services: {
        water: true,
        electricity: false,
        pet: false,
        bathroom: true,
        shower: true,
        internet: false,
        washClothes: true,
        pool: false,
        bar: false
      }
    },
    {
      title: 'Cangas de Onís',
      body: 'Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.',
      services: {
        water: true,
        electricity: false,
        pet: false,
        bathroom: true,
        shower: true,
        internet: false,
        washClothes: true,
        pool: false,
        bar: false
      }
    },
    {
      title: 'Cangas de Onís',
      body: 'Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.',
      services: {
        water: true,
        electricity: false,
        pet: false,
        bathroom: true,
        shower: true,
        internet: false,
        washClothes: true,
        pool: false,
        bar: false
      }
    }
  ]

  slider = document.querySelector('.slider-wrap')

  onClickLeft() {
    const slider = document.querySelector('.slider-wrap');
    slider!.scrollLeft -= 200;
  };

  onClickRight() {
    const slider = document.querySelector('.slider-wrap');
    slider!.scrollLeft += 200;
  };
}
