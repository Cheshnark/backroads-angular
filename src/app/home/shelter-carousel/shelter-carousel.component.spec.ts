import { ComponentFixture, TestBed, fakeAsync, tick } from '@angular/core/testing';

import { ShelterCarouselComponent } from './shelter-carousel.component';
import { By } from '@angular/platform-browser';

describe('ShelterCarouselComponent', () => {
  let component: ShelterCarouselComponent;
  let fixture: ComponentFixture<ShelterCarouselComponent>;
  let compiled: HTMLElement;

  // Befor the test block
  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [ShelterCarouselComponent]
    })
    .compileComponents();
  });
   
  beforeEach(async () => {
    fixture = TestBed.createComponent(ShelterCarouselComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.debugElement.nativeElement;
  });

  // Testing block
  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should match snapshot', () => {
    expect(compiled).toMatchSnapshot();
  })

  it('should call onClickLeft func', fakeAsync(() => {
    const spy = jest.spyOn(component, 'onClickLeft');

    const button = fixture.debugElement.nativeElement.querySelector('.left');
    button.click();
    tick();
    expect(spy).toHaveBeenCalled();
  }))

  it('should call onClickRight func', fakeAsync(() => {
    const spy = jest.spyOn(component, 'onClickRight');

    const button = fixture.debugElement.nativeElement.querySelector('.right');
    button.click();
    tick();
    expect(spy).toHaveBeenCalled();
  }))
});
