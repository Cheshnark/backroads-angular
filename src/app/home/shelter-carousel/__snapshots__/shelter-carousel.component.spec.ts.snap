// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShelterCarouselComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div>
    <div
      class="slider-wrap"
    >
      <div
        class="slider-main flex items-center gap-8"
      >
        <button
          class="left"
        >
          <app-chevron-left>
            <svg
              class="lucide lucide-chevron-left"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m15 18-6-6 6-6"
              />
            </svg>
          </app-chevron-left>
        </button>
        <app-shelter-cards>
          <div
            class="shelter-container"
          >
            <h3>
              Cangas de Onís
            </h3>
            <p>
              Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.
            </p>
            <div
              class="services-container flex flex-wrap gap-2"
            >
              <p>
                Watha
              </p>
              
              
              
              <p>
                BRoom
              </p>
              
              <p>
                Showaah
              </p>
              
              
              <p>
                Guas maschine
              </p>
              
              
              
            </div>
          </div>
        </app-shelter-cards>
        <app-shelter-cards>
          <div
            class="shelter-container"
          >
            <h3>
              Cangas de Onís
            </h3>
            <p>
              Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.
            </p>
            <div
              class="services-container flex flex-wrap gap-2"
            >
              <p>
                Watha
              </p>
              
              
              
              <p>
                BRoom
              </p>
              
              <p>
                Showaah
              </p>
              
              
              <p>
                Guas maschine
              </p>
              
              
              
            </div>
          </div>
        </app-shelter-cards>
        <app-shelter-cards>
          <div
            class="shelter-container"
          >
            <h3>
              Cangas de Onís
            </h3>
            <p>
              Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.
            </p>
            <div
              class="services-container flex flex-wrap gap-2"
            >
              <p>
                Watha
              </p>
              
              
              
              <p>
                BRoom
              </p>
              
              <p>
                Showaah
              </p>
              
              
              <p>
                Guas maschine
              </p>
              
              
              
            </div>
          </div>
        </app-shelter-cards>
        <app-shelter-cards>
          <div
            class="shelter-container"
          >
            <h3>
              Cangas de Onís
            </h3>
            <p>
              Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.
            </p>
            <div
              class="services-container flex flex-wrap gap-2"
            >
              <p>
                Watha
              </p>
              
              
              
              <p>
                BRoom
              </p>
              
              <p>
                Showaah
              </p>
              
              
              <p>
                Guas maschine
              </p>
              
              
              
            </div>
          </div>
        </app-shelter-cards>
        <app-shelter-cards>
          <div
            class="shelter-container"
          >
            <h3>
              Cangas de Onís
            </h3>
            <p>
              Ríos, vales, carreteras con mazo de curvas, que no curwas. No te preocupes por las cascadas, desprendimientos, hay curvas, velocidad, emoción, G-U-A-U, se te va la faking olla primo.
            </p>
            <div
              class="services-container flex flex-wrap gap-2"
            >
              <p>
                Watha
              </p>
              
              
              
              <p>
                BRoom
              </p>
              
              <p>
                Showaah
              </p>
              
              
              <p>
                Guas maschine
              </p>
              
              
              
            </div>
          </div>
        </app-shelter-cards>
        
        <button
          class="right"
        >
          <app-chevron-right>
            <svg
              class="lucide lucide-chevron-right"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="m9 18 6-6-6-6"
              />
            </svg>
          </app-chevron-right>
        </button>
      </div>
    </div>
  </div>
</div>
`;
